### PR TITLE
Cod order products images

### PR DESCRIPTION
**Pull request by gopalshimpi**
## Why?
**Integration**
* Integarted image api.
* Integarted COD API

## This change addresses the need by:
* Now COD(Cash on delivery) payment method is available for placing order.
* Images for product has been added.

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -58,14 +60,20 @@ export class AppComponent implements OnInit, OnDestroy {
     this.subscriptionList$.push(
       this.store.select(getAuthStatus).subscribe((data: boolean) => {
         if (data) {
-          this.orderSub$ = this.checkoutService.fetchCurrentOrder().subscribe();
+          this.orderSub$ = this.checkoutService.fetchCurrentOrder()
+            .subscribe(_ => { },
+              _ => {
+                localStorage.clear()
+                this.store.dispatch(this.authAction.logoutSuccess())
```
Comment (arjun289) [inline src/app/app.component.ts]: Why are you dispatching logout success on init?

File: src/app/checkout/actions/checkout.actions.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export class CheckoutActions {
   static BIND_ADDRESS = 'BIND_ADDRESS';
   static BIND_PAYMENT = 'BIND_PAYMENT';
   static GET_ORDER_PAYMENT_SUCCESS = 'GET_ORDER_PAYMENT_SUCCESS';
+  static SHIPPING_PREFERENCESS = 'SHIPPING_PREFERENCESS';
```
Comment (arjun289) [inline src/app/checkout/actions/checkout.actions.ts]: 'SHIPPING PREFERENCES'

File: src/app/checkout/actions/checkout.actions.ts [hunk 3]
```diff
@@ -123,5 +117,12 @@ export class CheckoutActions {
       payload: order
     }
   }
+
+  getShippingPreferencess(orderId: number, packages: Array<{}>) {
```
Comment (arjun289) [inline src/app/checkout/actions/checkout.actions.ts]: `getShippingPreferences`

File: src/app/checkout/address/address.component.ts [hunk 4]
```diff
@@ -73,7 +77,14 @@ export class AddressComponent implements OnInit, OnDestroy {
   }
 
   checkoutToPayment() {
-    this.router.navigate(['/checkout', 'payment']);
+    this.subscriptionList$.push(
+      this.store.select(getOrderId)
+        .subscribe(orderId => {
+          this.store.dispatch(this.checkOutActions.getShippingPreferencess(orderId, []))
+          // this.router.navigate(['/checkout', 'payment']);
```
Comment (arjun289) [inline src/app/checkout/address/address.component.ts]: any need for this commented code?

File: src/app/checkout/payment/payment-modes-list/cash-on-delivery/cash-on-delivery.component.html [hunk 5]
```diff
@@ -12,6 +12,10 @@
 	</div>
 </div>
 
-<div class="pay-btn-wrap">
-	<button (click)="onPay()" class="btn themebtnprimary">PAY ON DELIVERY</button>
-</div>
\ No newline at end of file
+<div class="pay-btn-wrap" *ngIf="!isPaymentAdded">
+		<button (click)="addPayment()" class="btn themebtnprimary">Continue</button>
```
Comment (arjun289) [inline src/app/checkout/payment/payment-modes-list/cash-on-delivery/cash-on-delivery.component.html]: indentation issue

File: src/app/checkout/payment/services/payment.service.ts [hunk 6]
```diff
@@ -70,6 +70,19 @@ export class PaymentService {
...
+
+  makeCodPayment(orderId: number) {
+    const params = this.buildCodPaymentJson(orderId);
+    const url = `api/v1/payment/cod_payment`
+    return this.http.post(url, params)
+      .pipe(
+        map(
+          res => { return res }
+        ),
+        error => { return error }
```
Comment (arjun289) [inline src/app/checkout/payment/services/payment.service.ts]: you have returned error from here, is it handled?

File: src/app/app.component.ts (hunk 1 above)
Comment (gopalshimpi) [reply src/app/app.component.ts]: It is to change the authstate in store. If token is expire.

**Pull request merged**